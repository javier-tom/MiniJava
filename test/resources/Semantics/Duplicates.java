class Main {
    public static void main(String[] args) {
        System.out.println(new B().run());
    }
}

// Duplicate of main class, definitely should cause error
class Main {
    int x;
}

class A {}

class A {
    // Duplicate, and should be reported as error
}

class B {
    public int run() {
        return 6;
    }

    // Duplicate, should cause error
    public int run() {
        return 7;
    }
}

class C {
    public int stuff() {
        int x;
        int x;
        x = 5;
        return x;
    }
}

class D {
    int x;
    int x;
}