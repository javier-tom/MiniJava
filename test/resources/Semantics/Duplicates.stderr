Error on line 8: duplicate class Main declared
Error on line 14: duplicate class A declared
Error on line 24: duplicate method run declared
Error on line 32: duplicate variable x delcared
Error on line 40: duplicate field x delcared